sdk-utils/broadcast/broadcast.go

```diff
@@ -270,16 +264,15 @@ func (b *pipelinedBroadcaster) Broadcast(ctx context.Context, msgs ...sdk.Msg) (
 
 	// need to be able to reorder msgs, so clone the msgs slice
 	retryMsgs := append(make([]sdk.Msg, 0, len(msgs)), msgs...)
-	err = b.retryPipeline.Push(
-		func() error {
+	err = b.retryPipeline.Push(ctx,
+		func(ctx context.Context) error {
 			response, err = b.broadcaster.Broadcast(ctx, retryMsgs...)
 			return err
 		},
 		func(err error) bool {
-			logger := b.logger.With("batch_size", len(retryMsgs))
 			i, ok := tryParseErrorMsgIndex(err)
 			if ok && len(retryMsgs) > 1 {
-				logger.Debug(fmt.Sprintf("excluding message at index %d due to error", i))
+				log.FromCtx(ctx).Debug(fmt.Sprintf("excluding message at index %d due to error", i))
```

Comment:
Due to the repetition of `log.FromCtx(ctx)`, it seems convenient to just pass `ctx` into `Debug` etc. instead where the logger gets created?

Reply:
the issue is that then it's impossible to define ctx free logging, i.e. `log.Debug(msg)`. We can of course decide that is the way to go

Reply:
Yeah, we could define an empty context helper bar to use in those cases. But if 95% of our logger usage will involve a ctx, then it seems more helpful this way. It also ensures that someone doesn't forget to pass in the context when logging

Reply:
good with both, slightly prefer to keep log.FromCtx, the interface looks cleaner